packages/design-system-react/src/components/button-base/ButtonBase.test.tsx

```diff
@@ -46,18 +46,29 @@ describe('ButtonBase', () => {
 
   it('shows loading state with loading text', () => {
     render(
-      <ButtonBase isLoading loadingText="Please wait...">
+      <ButtonBase
+        isLoading
+        loadingText="Please wait..."
+        loadingIconProps={{ 'data-testid': 'loading-spinner' }}
+      >
         Submit
       </ButtonBase>,
     );
-    expect(screen.getByRole('img')).toBeInTheDocument();
+    expect(screen.getByTestId('loading-spinner')).toBeInTheDocument();
```

Comment:
Updating tests that used the role of image for icons in button tests in favor of test ids